modules/listener/listener.bal

```diff
@@ -0,0 +1,154 @@
+        } else {
+            self.specificFolderOrFileId = EMPTY_STRING;
+            self.watchResponse = check startWatch(config.callbackURL, self.driveClient);
+            check getCurrentStatusOfDrive(self.driveClient, self.currentFileStatus);
+        }
+        self.channelUuid = self.watchResponse?.id.toString();
+        self.currentToken = self.watchResponse?.startPageToken.toString();
+        self.watchResourceId = self.watchResponse?.resourceId.toString();
+        log:print("Watch channel started in Google, id : " + self.channelUuid);
+    }
+    
+    public isolated function attach(http:Service s, string[]|string? name) returns error? {
+        return self.httpListener.attach(s, name);
+    }
+
+    public isolated function detach(http:Service s) returns error? {
+        return self.httpListener.detach(s);
+    }
+
+    public isolated function 'start() returns error? {
+        return self.httpListener.'start();
+    }
+
+    public isolated function gracefulStop() returns error? {
+        return ();
+    }
+
+    public isolated function immediateStop() returns error? {
+        return self.httpListener.immediateStop();
+    }
+
+    # Finding event type triggered and retrieve changes list.
+    # 
+    # + caller - The http caller object for responding to requests 
+    # + request - The HTTP request.
+    # + return - Returns error, if unsuccessful.
+    public function findEventType(http:Caller caller, http:Request request) returns @tainted error? {
+        log:print("< RECEIVING A CALLBACK <");
+        string channelID = check request.getHeader("X-Goog-Channel-ID");
+        string messageNumber = check request.getHeader("X-Goog-Message-Number");
+        string resourceStates = check request.getHeader("X-Goog-Resource-State");
+        string channelExpiration = check request.getHeader("X-Goog-Channel-Expiration");
+        if (channelID != self.channelUuid) {
+            fail error("Diffrent channel IDs found, Resend the watch request");
+        } else {
+            drive:ChangesListResponse[] response = check getAllChangeList(self.currentToken, self.driveClient);
+            foreach drive:ChangesListResponse item in response {
+                self.currentToken = item?.newStartPageToken.toString();
+                if (self.isWatchOnSpecificResource && self.isAFolder) {
+                    log:print("Folder watch response processing");
+                    check mapEventForSpecificResource(<@untainted> self.specificFolderOrFileId, <@untainted> item, 
+                    <@untainted> self.driveClient, <@untainted> self.eventService, <@untainted> self.currentFileStatus);
+                    check getCurrentStatusOfDrive(self.driveClient, self.currentFileStatus, self.specificFolderOrFileId);
+                } else if (self.isWatchOnSpecificResource && self.isAFolder == false) {
+                    log:print("File watch response processing");
```

Comment:
These should be DEBUG level logs. Otherwise there will be a log line per event when we use the connector.

Reply:
Please check all logs for above.

Reply:
Hi 
Changed to suitable log levels  in all occurrences
Thanks